rpcs3/Emu/System.cpp

```diff
@@ -961,6 +969,21 @@ game_boot_result Emulator::Load(const std::string& title_id, bool add_only, bool
 			{
 				m_path = rpcs3::utils::get_hdd0_dir();
 				m_path += std::string_view(argv[0]).substr(9);
+
+				constexpr auto game0_path = "/dev_hdd0/game/"sv;
+
+				if (argv[0].starts_with(game0_path) && !fs::is_file(vfs::get(argv[0])))
+				{
+					std::string title_id = argv[0].substr(game0_path.size());
+					title_id = title_id.substr(0, title_id.find_last_not_of('/'));
+
+					// Try to load game directory from list if available
+					if (auto node = (title_id.empty() ? YAML::Node{} : games[title_id]))
+					{
+						disc = node.Scalar();
+						m_path = disc + argv[0].substr(game0_path.size() + title_id.size());
+					}
+				}
```

Comment:
what does this do?

Reply:
Tries to find the digital game in games.yml

Reply:
This is for savestates, savestates save only guest path so it needs to deal with it by trying to find the game in places it can be.